### PR TITLE
Add init/setJSON functions 

### PR DESCRIPTION
**Pull request by jayahariv**
* Convert the nullable constructor/setters for data to non-nullable. In case of nil, use empty dictionary. (swift and objc)

Only for objc
* Add `initWithJSON:error` and `setJSON:error` functions to MutableDocument, MutableArray, MutableDictionary.
* Enhanced the `cbl_toCBLObject` dictionary category to handle in case of blob metadata in it.
* Add some unit tests. (Not optimized, since other PR might include some data to reuse)

Blob
* Handle Blob.content access on any Mutable Objects(JSON constructed)
* Skip install when there is digest associated with the metadata.

### CODE REVIEW

**Review by pasin — APPROVED** [review]

**Comment by jayahariv** [comment]
Oh my! did I added you as reviewer, it's too early to review, only in draft mode. Sorry! will add you back when its ready for review.

**Comment by jayahariv** [comment]
Failure on `testConflictResolverReturningBlobFromDifferentDB` might be related to the this change.

**Comment by pasin** [comment]
It this ready to re-review?

**Comment by jayahariv** [comment]
Yes, read to review

File: Objective-C/CBLMutableArray.mm [hunk 1]
```diff
@@ -228,14 +241,47 @@ - (void) insertDictionary: (nullable CBLDictionary*)value atIndex: (NSUInteger)i
...
+    FLSliceResult result = FLEncoder_Finish(enc, &flEerror);
+    if (!result.buf) {
+        CBLWarnError(Database, @"%@: Error decoding JSON: %d", self, flEerror);
+        convertError(flEerror, error);
+        return NO;
+    }
+    
+    FLArray array = FLValue_AsArray(FLValue_FromData(C4Slice(result), kFLTrusted));
+    CBL_LOCK(self.sharedLock) {
+        _array.clear();
```
Comment (pasin) [inline Objective-C/CBLMutableArray.mm:275]: Can we just replace _array with a new MArray? There is `MArray(MValue *mv, MCollection *parent)` that can be used to create a new Array with `array` and the parent from `_array.parent`.

File: Objective-C/CBLMutableDictionary.mm [hunk 2]
```diff
@@ -143,6 +153,39 @@ - (void) setData: (nullable NSDictionary<NSString*,id>*)data {
...
+    flEerror = {};
+    FLSliceResult result = FLEncoder_Finish(enc, &flEerror);
+    if (!result.buf) {
+        CBLWarnError(Database, @"%@: Error decoding JSON: %d", self, flEerror);
+        convertError(flEerror, error);
+        return NO;
+    }
+    
+    FLDict dict = FLValue_AsDict(FLValue_FromData(C4Slice(result), kFLTrusted));
+    _dict.clear();
```
Comment (pasin) [inline Objective-C/CBLMutableDictionary.mm:176]: Can we just replace _array with a new MDict? There is `MDict(MValue *mv, MCollection *parent)` that can be used to create a new Array with `dict` and the parent from `_array.parent`.

**Review by pasin — CHANGES_REQUESTED** [review]

File: Objective-C/CBLMutableArray.mm (hunk 1 above)
Comment (jayahariv) [reply Objective-C/CBLMutableArray.mm:275]: https://issues.couchbase.com/browse/CBL-1760

File: Objective-C/CBLMutableDictionary.mm (hunk 2 above)
Comment (jayahariv) [reply Objective-C/CBLMutableDictionary.mm:176]: https://issues.couchbase.com/browse/CBL-1760

**Review by pasin — APPROVED** [review]

**Pull request merged**